modules/caddytls/matchers.go

```diff
@@ -105,38 +106,35 @@ func (m MatchRemoteIP) Match(hello *tls.ClientHelloInfo) bool {
 	if err != nil {
 		ipStr = remoteAddr // weird; maybe no port?
 	}
-	ip := net.ParseIP(ipStr)
-	if ip == nil {
+	ipAddr, err := netip.ParseAddr(ipStr)
+	if err != nil {
 		m.logger.Error("invalid client IP addresss", zap.String("ip", ipStr))
 		return false
 	}
-	return (len(m.cidrs) == 0 || m.matches(ip, m.cidrs)) &&
-		(len(m.notCidrs) == 0 || !m.matches(ip, m.notCidrs))
+	return (len(m.cidrs) == 0 || m.matches(ipAddr, m.cidrs)) &&
+		(len(m.notCidrs) == 0 || !m.matches(ipAddr, m.notCidrs))
 }
 
-func (MatchRemoteIP) parseIPRange(str string) ([]*net.IPNet, error) {
-	var cidrs []*net.IPNet
+func (MatchRemoteIP) parseIPRange(str string) ([]*netip.Prefix, error) {
+	var cidrs []*netip.Prefix
 	if strings.Contains(str, "/") {
-		_, ipNet, err := net.ParseCIDR(str)
+		ipNet, err := netip.ParsePrefix(str)
 		if err != nil {
 			return nil, fmt.Errorf("parsing CIDR expression: %v", err)
 		}
-		cidrs = append(cidrs, ipNet)
+		cidrs = append(cidrs, &ipNet)
```

Comment:
Is there a good reason to append references here instead of the value? i.e. why not make the slice `[]netip.Prefix`?

Reply:
you have right, I will change it

Reply:
![image](https://user-images.githubusercontent.com/102859171/185227568-f7302ea9-f607-4c1d-ab53-9acee98643fb.png)
Other methods stop working after changes, so I not add it

Reply:
I think that it must be []*netip.Prefix type because it is expected returned type used by other methods

Reply:
when parseIPRange() returns []netip.Prefix it not working because m.cidrs requires []*netip.Prefix

Reply:
Why not change `m.cidrs` to `[]netip.Prefix`?

Reply:
okey it is working, i clear the commits and push the code